File: src/main/java/org/pcha/benthos/language/psi/BloblangFile.java
package org.pcha.benthos.language.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.lang.Language;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;
import org.pcha.benthos.language.BloblangFileType;
import org.pcha.benthos.language.BloblangLanguage;

public class BloblangFile extends PsiFileBase {

    public BloblangFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, BloblangLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return BloblangFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "Bloblang File";
    }
}

File: src/main/kotlin/com/github/pcha/bloblang/psi/BloblangFile.kt
package com.github.pcha.bloblang.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import com.github.pcha.bloblang.BloblangFileType
import com.github.pcha.bloblang.BloblangLanguage
import org.jetbrains.annotations.NotNull

class BloblangFile(viewProvider: FileViewProvider) :
    PsiFileBase(viewProvider, BloblangLanguage.INSTANCE) {
    override fun getFileType(): FileType {
        return BloblangFileType.INSTANCE
    }

    override fun toString(): String {
        return "Bloblang File"
    }
}
